Floofbot/Modules/RaidProtectionCommands.cs

```diff
@@ -0,0 +1,231 @@
+﻿using Discord;
+using Discord.Addons.Interactive;
+using Discord.Commands;
+using Discord.WebSocket;
+using Floofbot.Services.Repository;
+using Floofbot.Services.Repository.Models;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Floofbot.Modules
+{
+    [Summary("Raid protection configuration commands")]
+    [RequireUserPermission(GuildPermission.Administrator)]
+    [Group("raidconfig")]
+    public class RaidProtectionCommands : InteractiveBase
+    {
+        private FloofDataContext _floofDB;
+
+        public RaidProtectionCommands(FloofDataContext floofDB)
+        {
+            _floofDB = floofDB;
+        }
+        private Discord.Color GenerateColor()
+        {
+            return new Discord.Color((uint)new Random().Next(0x1000000));
+        }
+
+        private RaidProtectionConfig GetServerConfig(ulong server)
+        {
+            // checks if server exists in database and adds if not
+            var serverConfig = _floofDB.RaidProtectionConfigs.Find(server);
+            if (serverConfig == null)
+            {
+                _floofDB.Add(new RaidProtectionConfig
+                {
+                    ServerId = server,
+                    Enabled = false,
+                    ModChannelId = null,
+                    ModRoleId = null,
+                    MutedRoleId = null,
+                    BanOffenders = true,
+                    ExceptionRoleId = null
+                });
+                _floofDB.SaveChanges();
+                return _floofDB.RaidProtectionConfigs.Find(server);
+            }
+            else
+            {
+                return serverConfig;
+            }
+        }
+        private async Task<SocketRole> resolveRole(string input, SocketGuild guild, ISocketMessageChannel channel)
+        {
+            SocketRole role = null;
+            bool roleFound = false;
```

Comment:
This boolean is a bit redundant. You can just check if `role == null` because if role is set, it has been found.

Reply:
This is not the purpose of this boolean. It is used to check if multiple roles of the same name are found. Using role will overwrite the found role and will cause issues with multiple roles. By setting a flag when a role is found, i can check the state of that flag if another role is found. And if I just take the first instance of role, it will ignore duplicate roles and may not grab the right one

Reply:
Changed